qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/customers/CustomerDatabaseQueries.groovy

```diff
@@ -342,31 +342,83 @@ class CustomerDatabaseQueries {
         return result
     }
 
+    /**
+     * Add an affiliation to the database
+     *
+     * @param affiliation which needs to be added to the database
+     */
     void addAffiliation(Affiliation affiliation) throws DatabaseQueryException {
-        log.debug("Adding affiliation " + affiliation.organisation + " " + affiliation.addressAddition + "to database")
-
-        /*
-        if (affiliationExists) {
+        if (affiliationExists(affiliation)) {
             throw new DatabaseQueryException("Affiliation is already in the database.")
         }
         Connection connection = databaseSession.getConnection()
         connection.setAutoCommit(false)
 
         connection.withCloseable {it ->
             try {
-                int customerId = createNewCustomer(it, customer)
-                storeAffiliation(it, customerId, customer.affiliations)
+                createNewAffiliation(it, affiliation)
                 connection.commit()
             } catch (Exception e) {
                 log.error(e.message)
                 log.error(e.stackTrace.join("\n"))
                 connection.rollback()
                 connection.close()
-                throw new DatabaseQueryException("Could not create customer.")
+                throw new DatabaseQueryException("Could not create affiliation.")
             }
 
         }
-        */
+    }
+
+    private boolean affiliationExists(Affiliation affiliation) {
+        String query = "SELECT * FROM affililation WHERE organization = ? " +
+                "AND address_addition=? " +
+                "AND street=? " +
+                "AND country=? " +
+                "AND postal_code=? " +
+                "AND city=? " +
+                "AND category=?"
+
+        Connection connection = databaseSession.getConnection()
+
+        def affiliationAlreadyInDb = false
+
+        connection.withCloseable {
+            def statement = connection.prepareStatement(query)
+            statement.setString(1, affiliation.organisation)
+            statement.setString(2, affiliation.addressAddition)
+            statement.setString(3, affiliation.street)
+            statement.setString(4, affiliation.country)
+            statement.setString(5, affiliation.postalCode)
+            statement.setString(6, affiliation.city)
+            statement.setString(7, affiliation.category.toString())
+            statement.execute()
+            def result = statement.getResultSet()
+            affiliationAlreadyInDb = result.next()
```

Comment:
NIT: what is `result`? why do you do `.next()` on it? It would be nice to see it directly from looking at the code.

Reply:
For me it is explicit enough, I get what is meant here. :)